dolphin/alert_manager/trap_receiver.py

```diff
@@ -58,23 +62,73 @@ def _add_transport(self):
             config.addTransport(
                 self.snmp_engine,
                 udp.domainName,
-                udp.UdpTransport().openServerMode((self.trap_receiver_address, int(self.trap_receiver_port)))
+                udp.UdpTransport().openServerMode(
+                    (self.trap_receiver_address, int(self.trap_receiver_port)))
             )
         except Exception:
             raise ValueError("Port binding failed the provided port is in use.")
 
+    @staticmethod
+    def _extract_oid_value(var_bind):
+        """Extracts oid and value from var binds.
+        ex: varbind = (SNMPv2-MIB::snmpTrapOID.0 = SNMPv2-MIB::coldStart)
+        oid = snmpTrapOID
+        val = coldStart
+        """
+
+        # Separate out oid and value strings
+        var_bind_info = var_bind.prettyPrint()
+        var_bind_info = var_bind_info.split("=", 1)
+        oid = var_bind_info[0]
+        val = var_bind_info[1]
+
+        # Extract oid from oid string
+        # Example: get snmpTrapOID from SNMPv2-MIB::snmpTrapOID.0
+        oid = re.split('[::.]', oid)[2]
+
+        # Value can contain mib name also, if so, extract value from it
+        # Ex: get coldStart from SNMPv2-MIB::coldStart
+        if "::" in val:
+            val = re.split('[::]', val)[2]
+        val = val.strip()
+
+        return oid, val
+
     def _cb_fun(self, state_reference, context_engine_id, context_name,
-              var_binds, cb_ctx):
+                var_binds, cb_ctx):
         """Callback function to process the incoming trap."""
-        exec_context = self.snmp_engine.observer.getExecutionContext('rfc3412.receiveMessage:request')
-        LOG.info(
-            '#Notification from %s \n#ContextEngineId: "%s" \n#ContextName: "%s" \n#SNMPVER "%s" \n#SecurityName "%s"' % (
-            '@'.join([str(x) for x in exec_context['transportAddress']]), context_engine_id.prettyPrint(),
-            context_name.prettyPrint(), exec_context['securityModel'], exec_context['securityName']))
-        for oid, val in var_binds:
-            output = rfc1902.ObjectType(rfc1902.ObjectIdentity(oid), val).resolveWithMib(
-                self.mib_view_controller).prettyPrint()
-            LOG.info(output)
+        exec_context = self.snmp_engine.observer.getExecutionContext(
+            'rfc3412.receiveMessage:request')
+        LOG.info('#Notification from %s \n#ContextEngineId: "%s" '
+                 '\n#ContextName: ''"%s" \n#SNMPVER "%s" \n#SecurityName "%s" '
+                 % (
+                     '@'.join(
+                         [str(x) for x in exec_context['transportAddress']]),
+                     context_engine_id.prettyPrint(),
+                     context_name.prettyPrint(), exec_context['securityModel'],
+                     exec_context['securityName']))
+
+        var_binds = [rfc1902.ObjectType(rfc1902.ObjectIdentity(x[0]), x[1])
```

Comment:
As SNMP v3 trap may need specific security parameters, like username, password, auth protocol, to parse the incoming trap message, but those prarameters are not used as input parameter anywhere.

Reply:
As discussed, this will be handled by snmp engine internally once it is configured

Reply:
Could you plz add an issue to trace it?

Reply:
https://github.com/sodafoundation/SIM/issues - 31